Dummy data test test_get_random_value_on_first_of_month failure

Seen once only as far as I'm aware, and can't reproduce locally so far

Seen on:
https://github.com/opensafely-core/databuilder/actions/runs/3967909678/jobs/6800420732#step:4:457

```
  =================================== FAILURES ===================================
  ___________________ test_get_random_value_on_first_of_month ____________________
  Traceback (most recent call last):
    File "/home/runner/work/databuilder/databuilder/tests/unit/dummy_data/test_generator.py", line 125, in test_get_random_value_on_first_of_month
      assert len(set(values)) > 1, "dates are all identical"
  AssertionError: dates are all identical
  assert 1 > 1
   +  where 1 = len({datetime.date(2023, 1, 1)})
   +    where {datetime.date(2023, 1, 1)} = set([datetime.date(2023, 1, 1), datetime.date(2023, 1, 1), datetime.date(2023, 1, 1), datetime.date(2023, 1, 1), datetime.date(2023, 1, 1), datetime.date(2023, 1, 1), ...])
```

--- a/tests/unit/dummy_data/test_generator.py
+++ b/tests/unit/dummy_data/test_generator.py
@@ -121,7 +121,7 @@ def test_get_random_value_on_first_of_month(dummy_patient_generator):
         type=datetime.date,
         constraints=(Constraint.FirstOfMonth(),),
     )
-    values = [dummy_patient_generator.get_random_value(column_info) for _ in range(10)]
+    values = [dummy_patient_generator.get_random_value(column_info) for _ in range(100)]
     assert len(set(values)) > 1, "dates are all identical"
     assert all(value.day == 1 for value in values)
 

